docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,557 @@
+    * [#3369](https://github.com/napari/napari/pull/3369) (Add cancel actions to plugin manager)
+    * [#4074](https://github.com/napari/napari/pull/4074) (Use napari hub to list plugins)
+    * [#4520](https://github.com/napari/napari/pull/4520) (Rework how subprocesses are launched)
+
+Adding the necessary functionality to the plugin dialog, including:
+
+* Adapting the current `Installer` class to support not only `pip` but also `conda/mamba` in a
+  single code base.
+* Adding support for `install`, `uninstall`, and `update` commands.
+* Populating the listing with data obtained from the napari hub API [^napari-hub-api].
+* Updating the listing to filter and mark plugins based on their availability on conda-forge.
+
+Some UX/UI improvements that are not part of this milestone but could be implemented in future
+releases include:
+
+* Making the plugin dialog a side panel / side dockwidget (similar to VS Code
+  [^vscode-extensions-ui])
+* Allow for simultaneous install of plugins. Currently multiple plugins can be selected for
+  install, uninstall and update, but each on of these actions, are queued and run sequentially.
+* Allow advanced users to install packages from PyPI or other local sources via drag-and-drop. This
+  presents many risks that could irreversibly disrupt the installation, though, so it needs to be
+  considered very carefully.
+
+### Milestone 4: Enabling in-app napari version updates
+
+Our first attempts [^in-app-update-pr] to implement this consisted of a pop-up dialog that would
+notify the user about the available updates. Accepting the dialog would trigger the creation of a
+new environment (e.g. `napari-0.4.16`) alongside the existing one (e.g. `napari-0.4.15`). While it
+could be argued that this is not an "update" per se, but a new installation, it fulfills the same
+goal: the new version is available. If the user wants to get rid of the old one, an option can
+be offered to auto-clean on success, or after not using it for some configurable time.
+
+However, having several versions of napari coexisting might risk questions like "which napari
+version is the default one?", "which one will be in charge of updating napari?", "what if the
+update logic changes over time?". There possible workarounds, but from the user perspective it could
+get confusing very fast.
+
+For that reason, we decided to slightly redesign our approach to in-app updates. We will still use
+several environments to keep things tidy and performant (see "Detailed description" for Milestone 2
+for more details), but the update checks will mimic something closer to a local server-client model
+or, more accurately, a `napari-updater` process (name subject to change; other options include
+`napari-launcher` or  `napari-manager`) that can be queried by a running `napari` instance.
+
+* `napari-updater`: We will create a separate standalone package to handle updates outside of the
```

Comment:
in the interest of alternate/rejected approaches, I'd be curious to hear some thoughts on the advantage to `napari-updater` as opposed to just directing the user to the URL where they can download and install the new version.

Reply:
I believe the updater will try and install all their existing plugins as best it can and maintain their preferences/ settings. To me those are the essential features of update in place.

Reply:
Preferences and settings we already get for free (since it lives in user space).  So if it maintains plugins, that would be good to state explicitly

Reply:
That's how we initially designed the installers: each installation contains just one environment with a single napari version. If you want to "update", you get a notification with a link to the releases, and you run it again.

Some feedback we got then was that this requires too much effort, and doesn't really implement an updating mechanism. So we devised this other strategy where the napari installation handles several environments, each with a napari version, which brings some benefits to the table:

* Single location where we can track all the napari installations
* Potential support for napari flavors (a napari version + a collection of plugins)
* Close to no interactions required by the user

I'll add some lines to the document covering this, thanks!

---

With respect to user preferences, there's a risk we haven't covered yet (but we happened to discuss it in today's community meeting): conflicting preferences across napari installations due to the present plugins. A suggestion involves a mechanism for cascading preferences (system, user, napari version).

Reply:
Added in the last commit!